static/app/views/starfish/views/webServiceView/starfishView.tsx

```diff
@@ -1,177 +1,174 @@
+import {usePageError} from 'sentry/utils/performance/contexts/pageError';
 import {MutableSearch} from 'sentry/utils/tokenizeSearch';
 import usePageFilters from 'sentry/utils/usePageFilters';
-import withApi from 'sentry/utils/withApi';
 import {AVG_COLOR, THROUGHPUT_COLOR} from 'sentry/views/starfish/colours';
 import Chart, {useSynchronizeCharts} from 'sentry/views/starfish/components/chart';
 import MiniChartPanel from 'sentry/views/starfish/components/miniChartPanel';
 import {STARFISH_CHART_INTERVAL_FIDELITY} from 'sentry/views/starfish/utils/constants';
-import {DataTitles} from 'sentry/views/starfish/views/spans/types';
+import {useEventsStatsQuery} from 'sentry/views/starfish/utils/useEventsStatsQuery';
+import {DataTitles, getThroughputTitle} from 'sentry/views/starfish/views/spans/types';
 import {SpanGroupBar} from 'sentry/views/starfish/views/webServiceView/spanGroupBar';
 import {BaseStarfishViewProps} from 'sentry/views/starfish/views/webServiceView/starfishLanding';
 
 import EndpointList from './endpointList';
 
 export function StarfishView(props: BaseStarfishViewProps) {
-  const {organization, eventView} = props;
   const pageFilter = usePageFilters();
-  const theme = useTheme();
-  const [isLoading, setIsLoading] = useState(true);
+  const {selection} = pageFilter;
+  const [activeSpanGroup, setActiveSpanGroup] = useState<string | null>(null);
 
-  function renderCharts() {
-    const query = new MutableSearch([
+  function useRenderCharts() {
+    let query = new MutableSearch([
       'event.type:transaction',
       'has:http.method',
       'transaction.op:http.server',
     ]);
+    let dataset = DiscoverDatasets.METRICS;
+    let yAxis = ['tps()', 'http_error_count()', 'avg(transaction.duration)'];
+    let titles = [DataTitles.throughput, DataTitles.errorCount, DataTitles.avg];
+    if (activeSpanGroup) {
+      query = new MutableSearch([
+        'transaction.op:http.server',
+        `span.module:${activeSpanGroup}`,
+      ]);
+      dataset = DiscoverDatasets.SPANS_METRICS;
+      yAxis = ['sps()', 'http_error_count()', 'avg(span.self_time)'];
+      titles = [
+        getThroughputTitle(activeSpanGroup),
+        activeSpanGroup === 'http' ? DataTitles.errorCount : '--',
+        `Avg Duration of ${activeSpanGroup} ${
+          activeSpanGroup === 'db' ? 'Queries' : 'Spans'
+        }`,
+      ];
+    }
+
+    const {isLoading: loading, data: results} = useEventsStatsQuery({
+      eventView: EventView.fromNewQueryWithPageFilters(
+        {
+          name: '',
+          fields: [],
+          yAxis,
+          query: query.formatString(),
+          dataset,
+          version: 2,
+          interval: getInterval(selection.datetime, STARFISH_CHART_INTERVAL_FIDELITY),
+        },
+        selection
+      ),
+      enabled: true,
+      referrer: 'api.starfish-web-service.homepage-charts',
+      initialData: {},
+    });
+    useSynchronizeCharts([loading]);
```

Comment:
oh shoot, the rest of this is ready for review, but i still have to fix chart syncing